src/test/net_tests.cpp

```diff
@@ -300,13 +300,21 @@ BOOST_AUTO_TEST_CASE(cnetaddr_basic)
 
     // IPv6, scoped/link-local. See https://tools.ietf.org/html/rfc4007
     // We support non-negative decimal integers (uint32_t) as zone id indices.
-    // Test with a fairly-high value, e.g. 32, to avoid locally reserved ids.
+    // Test with a high value, e.g. 4096, to avoid any local ids in use.
     const std::string link_local{"fe80::1"};
-    const std::string scoped_addr{link_local + "%32"};
+    const std::string scoped_addr{link_local + "%4096"};
     BOOST_REQUIRE(LookupHost(scoped_addr, addr, false));
     BOOST_REQUIRE(addr.IsValid());
     BOOST_REQUIRE(addr.IsIPv6());
     BOOST_CHECK(!addr.IsBindAny());
+    const std::string addr_str{addr.ToString()};
+#ifdef MAC_OSX
+    // Expect macOS to return the address in this format, without zone ids.
+    BOOST_CHECK_EQUAL(addr_str, "fe80:0:0:0:0:0:0:1");
+#else
+    // Expect normal scoped address functionality on the other platforms.
+    BOOST_CHECK_EQUAL(addr_str, scoped_addr);
+#endif
```

Comment:
```suggestion
    BOOST_CHECK(addr.ToString().starts_with("fe80:0:0:0:0:0:0:1"));
```

Maybe just check the prefix if the suffix is unstable?